File: app/src/main/java/com/appchamp/wordchunks/models/realm/Word.java
package com.appchamp.wordchunks.models.realm;


import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;

public class Word extends RealmObject {

    @PrimaryKey
    private String id;
    @Index
    private String word;
    @Index
    private int state;  // 0 = not solved, 1 = solved
    @Index
    private int position;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }
}

File: app/src/main/java/com/appchamp/wordchunks/models/realm/Word.kt
package com.appchamp.wordchunks.models.realm

import io.realm.RealmObject
import io.realm.annotations.Index
import io.realm.annotations.Required


open class Word : RealmObject() {

    @Index
    @Required
    var word: String = ""
    @Index
    var state: Int = 0  // 0 = not solved, 1 = solved
    @Index
    var position: Int = 0

    fun getProperIndex(): String = when (position) {
        0 -> "1"
        1 -> "4"
        2 -> "2"
        3 -> "5"
        4 -> "3"
        5 -> "6"
        else -> throw IllegalArgumentException("Invalid index param value")
    }
}